### ISSUE TITLE
`launch.json` debugging in VS Code

### ISSUE THREAD

**Issue by garyking** [saltyshiomix/nextron#72]
Could you please provide the necessary config to be used in VS Code's `launch.json` file, to debug apps created by Nextron?

**Comment by garyking** [comment]
Okay, so this is what I've got:

```json
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "type": "node",
      "request": "launch",
      "name": "Next: Node",
      "runtimeExecutable": "${workspaceFolder}/node_modules/next/dist/bin/next",
      "runtimeArgs": ["-p", "8888", "./renderer"],
      "env": {
        "NODE_OPTIONS": "--inspect-brk"
      },
      "port": 9229,
      "console": "integratedTerminal"
    },
    {
      "name": "Debug Main Process",
      "type": "node",
      "request": "launch",
      "cwd": "${workspaceFolder}",
      "runtimeExecutable": "${workspaceFolder}/node_modules/.bin/electron",
      "args": [".", "8888"],
      "outputCapture": "std",
      "console": "integratedTerminal"
    }
  ],
  "compounds": [
    {
      "name": "Next: Full",
      "configurations": ["Next: Node", "Debug Main Process"]
    }
  ]
}
```

However, the `compound` debugger doesn't work, since it launches both at the same time, and if the `Debug Main Process` is ran too early, then it will fail, and therefore not launch Electron. 

So I have to manually start the Next server first, wait a second, then start Electron.

**Comment by garyking** [comment]
I'm also struggling to figure out how to set breakpoints in the React code, and make the VS Code debugger stop on them.

**Comment by lacymorrow** [comment]
Hi @garyking, I'm not familiar with VS code or the debugging used with it so I won't be able to help much there. If you are able to get things working properly we could look at pulling that config into the project for others to use.

As for debugging, the renderer processes have Chrome Devtools attached and available to use.

**Comment by frothywater** [comment]
I forked this repo and modified "lib/nextron-dev.ts":
`
const rendererPort = args['--port'] || 8888;
const remoteDebuggingPort = args['--remote-debugging-port'] || 5858;
const inspectPort = args['--inspect'] || 9292;
`
`
const startMainProcess = () => {
    mainProcess = spawn(
      'electron',
      [
        '.',
        `${rendererPort}`,
        `--remote-debugging-port=${remoteDebuggingPort}`,
        `--inspect=${inspectPort}`,
      ],
      {
        detached: true,
        ...spawnOptions,
      }
    );
    mainProcess.unref();
  };`

And below are my ".vscode/tasks.json" and ".vscode/launch.json" in my project:
`
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "npm",
      "script": "dev",
      "isBackground": true,
      "problemMatcher": {
        "owner": "custom",
        "pattern": {
          "regexp": ""
        },
        "background": {
          "beginsPattern": "started server",
          "endsPattern": "Debugger listening on"
        }
      },
      "label": "dev"
    }
  ]
}
`
`
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Nextron: Main",
      "type": "node",
      "request": "attach",
      "protocol": "inspector",
      "port": 9292,
      "timeout": 10000,
      "skipFiles": ["<node_internals>/**"],
      "sourceMapPathOverrides": {
        "webpack:///./~/*": "${workspaceFolder}/node_modules/*",
        "webpack:///./*": "${workspaceFolder}/*",
        "webpack:///*": "*"
      }
    },
    {
      "name": "Nextron: Renderer",
      "type": "chrome",
      "request": "attach",
      "port": 5858,
      "timeout": 10000,
      "urlFilter": "http://localhost:*",
      "webRoot": "${workspaceFolder}/app",
      "sourceMapPathOverrides": {
        "webpack:///./src/*": "${webRoot}/*"
      }
    }
  ],
  "compounds": [
    {
      "name": "Nextron: All",
      "preLaunchTask": "dev",
      "configurations": ["Nextron: Main", "Nextron: Renderer"]
    }
  ]
}
`

**Comment by lacymorrow** [comment]
@frothywater thanks for this! I'm not an expert with Typescript, are you able to make a PR? 
@saltyshiomix has built a wonderful framework and we're always looking for contributors!

**Comment by frothywater** [comment]
Sure! I'm always glad that I could be helpful!

**Comment by lacymorrow** [comment]
Please do :)

**Comment by saltyshiomix** [comment]
It is already merged at #87 , so I close it :)

**Comment by bystepii** [comment]
I tried debugging in VS Code but it is not working:

![image](https://user-images.githubusercontent.com/9701284/211100974-c0f0ca0d-8d27-43d6-8cf1-88b77b15cd8a.png)
